testsuite/MDAnalysisTests/formats/test_libmdaxdr.py

```diff
@@ -21,376 +21,495 @@
 #
 from __future__ import print_function, absolute_import
 
-from nose.tools import raises
-from numpy.testing import assert_equal, assert_array_equal
-from numpy.testing import assert_array_almost_equal
+import numpy as np
+
 from numpy.testing import assert_almost_equal
+from numpy.testing import assert_array_almost_equal
+from numpy.testing import assert_array_equal
 
 from MDAnalysis.lib.formats.libmdaxdr import TRRFile
-from MDAnalysisTests.datafiles import TRR_single_frame
 from MDAnalysisTests.datafiles import TRR_multi_frame
+from MDAnalysisTests.datafiles import TRR_single_frame
 
 from MDAnalysis.lib.formats.libmdaxdr import XTCFile
-from MDAnalysisTests.datafiles import XTC_single_frame
 from MDAnalysisTests.datafiles import XTC_multi_frame
+from MDAnalysisTests.datafiles import XTC_single_frame
 
-from MDAnalysisTests.tempdir import run_in_tempdir
-import numpy as np
-
+import pytest
 
-class XDRFormatBaseTest(object):
+XTC_OFFSETS = np.array([0, 104, 208, 312, 416, 520, 624, 728, 832, 936])
+TRR_OFFSETS = np.array([0, 480, 960, 1440, 1920, 2400, 2880, 3360, 3840, 4320])
 
-    def test_n_atoms(self):
-        f = self.xdrfile(self.single_frame)
-        assert_equal(f.n_atoms, 10)
 
-    @raises(IOError)
-    def test_raise_not_existing(self):
-        self.xdrfile('foo')
+def test_xtc():
+    with XTCFile(XTC_single_frame) as f:
+        assert f.n_atoms == 10
+        xyz, box, step, time, prec = f.read()
+        # xtc only saves with 3 decimal places precision
+        assert_array_almost_equal(xyz.flat, np.arange(30), decimal=3)
+        assert_array_almost_equal(box, np.eye(3) * 20, decimal=3)
+        assert step == 0
```

Comment:
I don't like having multiple asserts in one test function, it makes it harder to debug.  Eg, if `xyz.flat` stopped working, I'd also want to know if `box` was working too, to help me try and locate the problem.  This test wouldn't give me the breakdown of what is broken.

I know you're not a fan of classes, but ..

```python
class TestXTC(object):
    @staticmethod
    @pytest.fixture()
    def XTC():
        with XTCFile(XTC_single_frame) as f:
            yield f

    def test_n_atoms(self, XTC):
        assert XTC.n_atoms == 10

    def test_xyc(self, XTC):
        etc
```
This is a lot nicer imo